internal/db/schema/migrations/oss/postgres/59/01_ldap.up.sql

```diff
@@ -0,0 +1,350 @@
+      check(length(certificate_key) > 0),
+  certificate_key_hmac bytea not null
+      constraint certificate_key_hmac_must_not_be_empty
+          check(length(certificate_key_hmac) > 0),
+  key_id text not null
+    constraint kms_data_key_version_fkey
+      references kms_data_key_version (private_id)
+      on delete restrict
+      on update cascade
+  );
+  comment on table auth_ldap_client_certificate is
+    'auth_ldap_client_certificate entries contains a client certificate that a '
+    'auth_ldap_method uses for mTLS when connecting to an LDAP server. '
+    'An auth_ldap_method can have 0 or 1 client certificates.';
+
+create table auth_ldap_bind_credential (
+  create_time wt_timestamp,
+  auth_ldap_method_id wt_public_id primary key
+      constraint auth_ldap_method_fkey
+        references auth_ldap_method (public_id)
+        on delete cascade
+        on update cascade,
+  dn text not null
+    constraint dn_too_short
+      check (length(trim(dn)) > 0)
+    constraint dn_too_long
+      check (length(trim(dn)) < 2049),
+  password bytea not null, -- encrypted password0
+  password_hmac text not null
+    constraint password_hmac_not_empty
+    check(length(trim(password_hmac)) > 0),
+  key_id text not null
+    constraint kms_data_key_version_fkey
+      references kms_data_key_version (private_id)
+      on delete restrict
+      on update cascade
+);
+comment on table auth_ldap_bind_credential is 
+'auth_ldap_bind_credentials entries allow Boundary to bind (aka authenticate) using '
+'the provided credentials when searching for the user entry used to authenticate.';
+
+create table auth_ldap_account (
+  public_id wt_public_id primary key,
+  auth_method_id wt_public_id not null,
+  -- NOTE(mgaffney): The scope_id type is not wt_scope_id because the domain
+  -- check is executed before the insert trigger which retrieves the scope_id
+  -- causing an insert to fail.
+  scope_id text not null,
+  name wt_name,
+  description wt_description,
+  create_time wt_timestamp,
+  update_time wt_timestamp,
+  version wt_version,
+  login_name text not null
+    constraint login_name_must_be_lowercase
+      check(lower(trim(login_name)) = login_name)
+    constraint login_name_must_not_be_empty
+      check(length(trim(login_name)) > 0),
+  foreign key (scope_id, auth_method_id)
+    references auth_ldap_method (scope_id, public_id)
+    on delete cascade
+    on update cascade,
+  foreign key (scope_id, auth_method_id, public_id)
+    references auth_account (scope_id, auth_method_id, public_id)
+    on delete cascade
+    on update cascade,
+  constraint auth_ldap_account_auth_method_id_name_uq
+    unique(auth_method_id, name),
+  constraint auth_ldap_account_auth_method_id_login_name_uq
+    unique(auth_method_id, login_name),
+  constraint auth_ldap_account_auth_method_id_public_id_uq
+    unique(auth_method_id, public_id)
+);
+
+create trigger insert_auth_account_subtype before insert on auth_ldap_account
```

Comment:
Do we need `update` and `delete` triggers for `auth_account_subtype`?

Reply:
we definitely need a delete trigger. I don't believe we need an update trigger for the sub_type